devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -490,7 +490,13 @@ void prepare() throws Exception {
             for (Map.Entry<Object, Object> e : System.getProperties().entrySet()) {
                 devModeContext.getSystemProperties().put(e.getKey().toString(), (String) e.getValue());
             }
+
             devModeContext.getBuildSystemProperties().putAll((Map) project.getProperties());
+
+            //  this is a minor hack to allow ApplicationConfig to be populated with defaults
+            devModeContext.getBuildSystemProperties().putIfAbsent("quarkus.application.name", project.getArtifactId());
```

Comment:
Should it resolve to `project.name` first and fallback to `artifactId`?

Reply:
Does the former exist for Gradle as well?

Also note that we already do the same thing for the regular build, it was just missing from dev mode

Reply:
> Does the former exist for Gradle as well?

I have no idea, but now that you mention the `QuarkusDev` task in the Gradle plugin needs to be updated too, no?

> Also note that we already do the same thing for the regular build, it was just missing from dev mode

:+1:

Reply:
It does yes, I was hoping you would take that up once this is merged 😉